go/database/mpt/archive_trie_test.go

```diff
@@ -2586,8 +2620,9 @@ func TestArchiveTrie_RestoreBlockHeight(t *testing.T) {
 			)
 		},
 		"no_close": func(archive *ArchiveTrie) error {
+			// close the inner forest to release resources
 			// In order to allow the recovery to access the directory, the lock needs to be released
-			return archive.head.(*MptState).lock.Release()
+			return errors.Join(archive.head.(*MptState).lock.Release(), archive.forest.Close())
```

Comment:
This test is actually aiming to recover an archive that has not been properly close. By closing it now, the test setup is changed.

To fix this, you could pass `t *testing.T` to the this test setup function and register the close operation as a cleanup step.

Reply:
ok, updated. I think this version was also correct, because it closes only the inner `forest`, and the archive stays updated, but right, using the test clean-up phase makes it cleaner.